lib/Doctrine/ORM/Internal/CommitOrderCalculator.php

```diff
@@ -118,47 +98,45 @@ public function sort()
      */
     private function visit(Vertex $vertex): void
     {
+        if ($vertex->state === VertexState::IN_PROGRESS) {
+            // This node is already on the current DFS stack. We've found a cycle!
+            throw new CycleDetectedException();
```

Comment:
An exception should have a message.

Reply:
Yes, will need to think about this a bit more. Probably the message should convey the exact cycle, ie which chain of entity associations is causing a foreign key loop

Reply:
Fixed